server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -287,12 +363,21 @@ public boolean isSourceEnabled() {
      * Generally used to handle unmapped fields in the context of sorting.
      */
     public MappedFieldType buildAnonymousFieldType(String type) {
-        final Mapper.TypeParser.ParserContext parserContext = mapperService.parserContext();
+        return buildFieldType(type, "__anonymous_" + type, Collections.emptyMap(), mapperService.parserContext(), indexSettings);
+    }
+
+    private static MappedFieldType buildFieldType(
+        String type,
+        String field,
+        Map<String, Object> node,
+        Mapper.TypeParser.ParserContext parserContext,
+        IndexSettings indexSettings
+    ) {
         Mapper.TypeParser typeParser = parserContext.typeParser(type);
         if (typeParser == null) {
             throw new IllegalArgumentException("No mapper found for type [" + type + "]");
         }
-        final Mapper.Builder builder = typeParser.parse("__anonymous_" + type, Collections.emptyMap(), parserContext);
+        Mapper.Builder builder = typeParser.parse(field, node, parserContext);
         Mapper mapper = builder.build(new ContentPath(1));
```

Comment:
I was wondering about this line: is it good also for runtime fields?

Reply:
The `new ContentPath(1)` bit?

Reply:
yes

Reply:
I looked around a bit and I'm not sure why anonymous fields use `1` here instead of `0` or `123` or any other number. I think we don't end up in the positions of building a path to add anyway. I'll do some more experimenting to see if I can make something "funny" if this is set to `0`.